### ISSUE TITLE
Enabler: Rebuild SideNav for NCIDS Side Nav

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#3344]
* The side nav root should have a `data-menu-id` attribute with the ID of that Drupal site section

> **ESTIMATE** 2

### InVision: 
  * DSM link: N/A
  * Desktop mockup: https://publicis.invisionapp.com/share/SY11BOIJQK4T#/454941204_Sidenav

### Acceptance Criteria for Behaviors 
*If a component changes between breakpoints, or things show/hide, that gets captured here* 

```gherkin
Scenario: User is on an L2 and lower of Cancer.gov that has no child pages
  Given they are on a screen 1024px and larger
    And the page has no child pages
  Then the left nav appears
    And the current page is highlighted with an indicator and has an aria-current="page" attribute
    And any siblings of the page are exposed
    And all ancestors (parents and parent's sibilings, grandparents and grandparent's siblings) appear until reaching site section root
    And the ancestors are highlighted
```

```gherkin
Scenario: User is on an L2 and lower and a sibling page has child pages
  Given they are on a screen 1024px and larger
    And a sibling page has child pages
  Then the left nav appears
    And the current page is highlighted with an indicator and has an aria-current="page" attribute
    And its sibling with child pages has a right arrow (>) icon appearing to the right of the label of the page 
    And the whole line is clickable (label and icon)
    And its sibling without child pages does not have a right arrow
```

```gherkin
Scenario: User is on a section nav root
  Given they are on a screen 1024px and larger
  Then the left nav appears
    And the current page is highlighted with an indicator  and has an aria-current="page" attribute
    And any children of the page are exposed
  When a child page is clicked on OR its corresponding > icon is clicked (the whole line is clickable - no split buttons)
  Then the page loads to that child page
    And the current page is highlighted with an indicator
    And any children of the page are exposed
    And any siblings of the page are exposed
    And all ancestors (parents and parent's sibilings, grandparents and grandparent's siblings) appear until reaching site section root
    And the ancestors are highlighted
```
**TODO** Add a scenario for when you are on a page like Cancer Treatment > Side Effects > Nausea, where Nausea does not show in the navigation. Side Effects should ONLY be an ancestor and not be active. There should be no aria-current="page" attribute. Also see [Joe's comment below](https://github.com/NCIOCPL/cgov-digital-platform/issues/3344#issuecomment-1161734971) on what the analytics activeLevel is. 

As a content editor
if I change the name of a site section in the side nav
Then that page and all child pages should have the side nav updated with the new name

As a content editor
If I change a site section to hide in left nav
Then that page and all child pages should be removed from the site nav

* Draw Side Nav
* Caching Crap
* Getting Data to Mobile Menu JS
* Don't break DCEG Connect

### Acceptance Criteria for Analytics
``` gherkin
Scenario: User clicks on a link in the left navigation
Given a user is on any page of cancer.gov
When the user clicks on a link within the side nav section of the page
Then there should be an NCIDataLayer Other event 
  And the event has the name "SideNav:LinkClick"
  And the event has the the linkname "SideNav:LinkClick"
  And this event should have the following data:
      | key            | value                                                                                                                        |
      | linkText       | Text of the link clicked                                                                                                     |
      | location       | SideNav                                                                                                                      |
      | listNumber     | Number in list of links within column of link clicked  (Treat the navigation as a flattened list)                            |
      | clickedLevel   | The value of the hierarchy of the link clicked. ie: (The number of `<ul>` parents preceded by the letter `L`)                |
      |                | If the top-level link is clicked, return L1. If the second link is clicked, return L2.                                       |
      | activeLevel    | The value of the hierarchy of the current page. Find the current active item and figure out the parent ULs preceded with `L` |
      | sideNavHeading | The name of the Section Nav Root (i.e. the first <li> in the nav)                                                            |
```

**Comment by bryanpizzillo** [comment]
@welshja or @nicolecrom What is the activeLevel for a page like https://ncigovcdode471.prod.acquia-sites.com/about-cancer/treatment/side-effects/anemia?
(None of the side effects display in the sidenav, only the parent Side Effects page)

**Comment by welshja** [comment]
@bryanpizzillo - please match active level to what is shown in the navigation itself as "active". This will apply even if it doesn't exactly match the page a user is on (like in the example you shared).

**Comment by sarinapadilla** [comment]
Closing this ticket as it was fixed by PR #3344!
